File: services/assesment-service/src/main/java/sgptt/assessvc/assesmentservice/repository/ActivityRepository.java
package sgptt.assessvc.assesmentservice.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import sgptt.assessvc.assesmentservice.entity.Activity;

@Repository
public interface ActivityRepository extends JpaRepository<Activity, Integer> {

}

File: services/protocols-service/src/main/kotlin/com/sgptt/protocolsservice/repository/ActivityRepository.kt
package com.sgptt.protocolsservice.repository

import com.sgptt.protocolsservice.model.ActivityName
import com.sgptt.protocolsservice.repository.entity.Activity
import org.springframework.data.jpa.repository.JpaRepository

interface ActivityRepository : JpaRepository<Activity, Long> {
	fun findByActivity(activity: ActivityName): Activity
}